### ISSUE TITLE
gccemacs native-comp failed.

### ISSUE THREAD

**Issue by DogLooksGood** [polymode/polymode#285]
polymode can't be compiled successfully in gccemacs. 

```
Compiling /home/tianshu/.emacs.d/straight/build/polymode/polymode-core.el...
Debugger entered--Lisp error: (invalid-function #<subr or>)
  or("weaving")
  cl-typep("weaving" or)
  #f(compiled-function (type) #<bytecode 0x162ac0dd07e53dbe>)(or)
  cl-some(#f(compiled-function (type) #<bytecode 0x162ac0dd07e53dbe>) (or))
  #f(compiled-function (s) #<bytecode 0x3c746cd0ade226a>)(#s(comp-cstr :typeset (or) :valset nil :range nil :neg nil))
  cl-every(#f(compiled-function (s) #<bytecode 0x3c746cd0ade226a>) (#s(comp-cstr :typeset (or) :valset nil :range nil :neg nil)))
  comp-cstr-intersection-homogeneous(#s(comp-cstr :typeset nil :valset nil :range nil :neg nil) #s(comp-mvar :typeset nil :valset ("weaving" "exporting") :range nil :neg nil :id 23460883616494 :slot 15) #s(comp-mvar :typeset nil :valset ("weaving" "exporting") :range nil :neg nil :id 23460883616494 :slot 15) #s(comp-cstr :typeset (or) :valset nil :range nil :neg nil))
  apply(comp-cstr-intersection-homogeneous #s(comp-cstr :typeset nil :valset nil :range nil :neg nil) (#s(comp-mvar :typeset nil :valset ("weaving" "exporting") :range nil :neg nil :id 23460883616494 :slot 15) #s(comp-mvar :typeset nil :valset ("weaving" "exporting") :range nil :neg nil :id 23460883616494 :slot 15) #s(comp-cstr :typeset (or) :valset nil :range nil :neg nil)))
  comp-cstr-intersection-no-mem(#s(comp-mvar :typeset nil :valset ("weaving" "exporting") :range nil :neg nil :id 23460883616494 :slot 15) #s(comp-mvar :typeset nil :valset ("weaving" "exporting") :range nil :neg nil :id 23460883616494 :slot 15) #s(comp-cstr :typeset (or) :valset nil :range nil :neg nil))
  comp-cstr-intersection(#s(comp-mvar :typeset (t) :valset nil :range nil :neg nil :id 23460885101028 :slot 15) #s(comp-mvar :typeset nil :valset ("weaving" "exporting") :range nil :neg nil :id 23460883616494 :slot 15) #s(comp-mvar :typeset nil :valset ("weaving" "exporting") :range nil :neg nil :id 23460883616494 :slot 15) #s(comp-cstr :typeset (or) :valset nil :range nil :neg nil))
  comp-fwprop-insn((assume #s(comp-mvar :typeset (t) :valset nil :range nil :neg nil :id 23460885101028 :slot 15) (and #s(comp-mvar :typeset nil :valset ("weaving" "exporting") :range nil :neg nil :id 23460883616494 :slot 15) #s(comp-mvar :typeset nil :valset ("weaving" "exporting") :range nil :neg nil :id 23460883616494 :slot 15) #s(comp-cstr :typeset (or) :valset nil :range nil :neg nil))))
  #f(compiled-function (--cl-var-- b) #<bytecode 0x80914955a1a8eba>)(bb_29 #s(comp-block-lap :name bb_29 :insns ... :closed t :in-edges ... :out-edges ... :idom ... :df #<hash-table eql 1/65 0x155669aa7125> :post-num 14 :final-frame ... :sp 15 :addr 138 :non-ret-insn nil :no-ret nil))
  comp-fwprop*()
  #<subr F616e6f6e796d6f75732d6c616d626461_anonymous_lambda_111>("F706d2d2d70726f636573732d696e7465726e616c_pm__proc..." #s(comp-func-l :name pm--process-internal :c-name "F706d2d2d70726f636573732d696e7465726e616c_pm__proc..." :byte-func #f(compiled-function (processor from to ifile &optional callback quote) #<bytecode -0xa71f8f7cb12f64d>) :doc "\n\n(fn PROCESSOR FROM TO IFILE &optional CALLBACK Q..." :int-spec nil :lap ((byte-constant object-of-class-p . 6) (byte-stack-ref . 6) (byte-constant pm-exporter . 7) (byte-call . 2) (byte-dup) (byte-goto-if-nil TAG 28 . 7) (byte-stack-ref . 5) (byte-goto-if-nil TAG 17) (byte-stack-ref . 4) (byte-goto-if-not-nil TAG 38 . 7) (TAG 17) (byte-constant error . 8) (byte-constant "For exporter both FROM and TO must be supplied (fr..." . 9) (byte-stack-ref . 7) (byte-stack-ref . 7) (byte-call . 3) (byte-discard) (byte-goto TAG 38 . 7) (TAG 28 . 7) (byte-stack-ref . 5) (byte-goto-if-not-nil TAG 38 . 7) (byte-constant error . 8) (byte-constant "For weaver FROM must be supplied (from: %s)" . 10) (byte-stack-ref . 7) (byte-call . 2) (byte-discard) (TAG 38 . 7) (byte-dup) (byte-goto-if-nil TAG 52 . 7) (byte-constant pm--selector . 11) (byte-stack-ref . 7) (byte-constant :from . 12) (byte-stack-ref . 8) (byte-call . 3) (byte-goto TAG 59 . 8) (TAG 52 . 7) (byte-constant pm--selector . 11) (byte-stack-ref . 7) (byte-constant :from-to . 13) (byte-stack-ref . 8) (byte-call . 3) (TAG 59 . 8) (byte-stack-ref . 1) (byte-goto-if-nil-else-pop TAG 70 . 9) (byte-constant pm--selector . 11) (byte-stack-ref . 8) (byte-constant :to . 14) (byte-stack-ref . 8) (byte-call . 3) (TAG 70 . 9) ...) :ssa-status t :frame-size 24 :blocks #<hash-table eq 60/65 0x1556699ac98b> :lap-block #<hash-table equal 0/65 0x1556699ad1b3> :edges-h #<hash-table eql 78/97 0x155669a797ed> :block-cnt-gen #f(compiled-function () #<bytecode 0x1f53ba4c33a6>) :edge-cnt-gen #f(compiled-function () #<bytecode 0x1f53ba4c28e2>) :has-non-local nil :array-h #<hash-table eql 1/65 0x1556699ac9ab> :speed 2 :pure nil :type #s(comp-mvar :typeset nil :valset (nil) :range nil :neg nil :id nil :slot nil) :args #s(comp-args :min 4 :max 6)))
  comp-fwprop(nil)
  #f(compiled-function (pass) #<bytecode -0xa70570367f1ef25>)(comp-fwprop)
  mapc(#f(compiled-function (pass) #<bytecode -0xa70570367f1ef25>) (comp-spill-lap comp-limplify comp-fwprop comp-call-optim comp-ipa-pure comp-add-cstrs comp-fwprop comp-dead-code comp-tco comp-fwprop comp-remove-type-hints comp-final))
  comp--native-compile("/home/tianshu/.emacs.d/straight/build/polymode/pol..." t)
  eval-buffer(#<buffer  *load*> nil "/tmp/emacs-async-comp-polymode-core-lgHMW5.el" nil t)  ; Reading at buffer position 3635
  load-with-code-conversion("/tmp/emacs-async-comp-polymode-core-lgHMW5.el" "/tmp/emacs-async-comp-polymode-core-lgHMW5.el" nil t)
  load("/tmp/emacs-async-comp-polymode-core-lgHMW5.el" nil t)
  command-line-1(("-l" "/tmp/emacs-async-comp-polymode-core-lgHMW5.el"))
  command-line()
  normal-top-level()

Compilation finished.
```

**Comment by bryce-carson** [comment]
I was encountering this issue, or a very closely related one, just yesterday. This is related to polymode/poly-R#22. When I tried using anything Polymode I would get:

1. `poly-R.el:58:1: Error: Symbol’s value as variable is void: keymap`
2. **_After changing the order of packages_** so that poly-noweb is loaded before poly-R or poly-markdown, I got `Debugger entered--Lisp error: (invalid-function #<subr or>)`.

I was able to resovle it by updating to a new nativecomp build including [the fix for #46912 in Emacs itself](https://github.com/emacs-mirror/emacs/commit/77ec25122c9a7855657de564101c9fe763b91013). The patch for that bug was published 2021-03-05, so you need to be running a very recent build of nativecomp.

If you're running Fedora 33, by chance, you could try my forked build from here: [Copr bacarson/emacs-pgtk-nativecomp](https://copr.fedorainfracloud.org/coprs/bacarson/emacs-pgtk-nativecomp/).

If your build is newer than 2021-03-05 then something else might be amiss.

**Comment by DogLooksGood** [comment]
Yes, I think update to latest native-comp solve the problem. 

Thank you very much! @bryce-carson
